tasks/files/snapshot.py

```diff
@@ -2076,6 +1855,107 @@ def validate_snapset_json(cmd, snapset, verify_only):
     return rc, message, snapset_json
 
 
+def get_mount_json_from_args(args):
+    volume_list = []
+    args_json = {}
+    volume = {}
+
+    if args.suffix:
+        args_json["name"] = args.suffix
+
+    if hasattr(args, "create"):
+        volume["create"] = args.create
+    if hasattr(args, "origin"):
+        volume["origin"] = args.origin
+    if hasattr(args, "verify"):
+        volume["verify"] = args.verify
+    if hasattr(args, "mountpoint"):
+        volume["mountpoint"] = args.mountpoint
+    if hasattr(args, "fstype"):
+        volume["fstype"] = args.fstype
+    if hasattr(args, "options"):
+        volume["options"] = args.options
+    if hasattr(args, "suffix"):
+        volume["suffix"] = args.suffix
+    if hasattr(args, "logical_volume"):
+        volume["lv"] = args.logical_volume
+    if hasattr(args, "volume_group"):
+        volume["vg"] = args.volume_group
+
+    volume_list.append(volume)
+    args_json["volumes"] = volume_list
+
+    return SnapshotStatus.SNAPSHOT_OK, "", json.dumps(args_json)
```

Comment:
The arguments are converted to a list of dict, then into a JSON string, then eventually back into a list of dict?

Reply:
How about if I move the checks up a level?  Something like this?  Then pass a dict to the "_cmd" functions...

```

    args = parser.parse_args()
    if args.vg_include:
        VG_INCLUDE = re.compile(args.vg_include)
    else:
        VG_INCLUDE = None

    if hasattr(args, "set_json"):
        result, snapset_dict = validate_snapset_json(
            get_command_const(args.operation), args.set_json, False
        )
    else:
        result, snapset_dict = validate_args(args)


    if result["return_code"] == SnapshotStatus.SNAPSHOT_OK:
        result = args.func(args, snapset_dict)
    print_result(result)

    sys.exit(result["return_code"])
```

Reply:
fixed